[docs] Add more info on query params (#2811)
sveltejs/kit: documentation/docs/03-loading.md

> Code called inside `load` blocks:
>

> > - should use the SvelteKit-provided [`fetch`](#loading-input-fetch) wrapper rather than using the native fetch as the latter will make the request twice each during SSR and hydration

Reviewer: I like that this explains why, but it could be said a bit more clearly as it somewhat sounds like twice during SSR and twice during hydration. also not sure why you'd take fetch out of backticks. 
```suggestion
> - should use the SvelteKit-provided [`fetch`](#loading-input-fetch) wrapper rather than using the native `fetch` because SvelteKit's version has built-in hydration support to avoid making a second request on the client.
```

> > - should use the SvelteKit-provided [`fetch`](#loading-input-fetch) wrapper rather than using the native fetch as the latter will make the request twice each during SSR and hydration

Reviewer: I'm not sure about this change. the reason to do it is given right above this. now the reason is being repeated, but only a partial explanation is being given

it also could be said a bit more clearly as it somewhat sounds like twice during SSR and twice during hydration. also not sure why you'd take fetch out of backticks

Author: Ok, will remove this change.

> > - should use the SvelteKit-provided [`fetch`](#loading-input-fetch) wrapper rather than using the native `fetch`. Although, using native `fetch` will work, it will cause the API server to be called twice - during SSR and hydration

Reviewer 2: How about

"... using the native `fetch` as the latter will make the request twice each during SSR and hydration"

Author: yes, this sound better

#### page

> `page` is a `{ host, path, params, query }` object where `host` is the URL's host, `path` is its pathname, `params` is derived from `path` and the route filename, and `query` is an instance of [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). Mutating `page` does not update the current URL, rather, it should be done by means of navigation using `goto`.

Author: @benmccann @bluwy can we add this information here?

Reviewer: this sounds reasonable. I wonder if an even better solution than documenting it might be to make the store exposed to the user readonly? https://github.com/sveltejs/svelte/pull/6518

Reviewer 3: I think `page` is referred here as an object rather than a store, but if it was imported from `$app/stores`, it's already a [readable store](https://kit.svelte.dev/docs#modules-$app-stores). So we would need to freeze the object, but I'm not sure if it's something we want to do. The change looks good to me though.

Reviewer 4: ```suggestion
`page` is a `{ host, path, params, query }` object where `host` is the URL's host, `path` is its pathname, `params` is derived from `path` and the route filename, and `query` is an instance of [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams). Mutating `page` does not update the current URL; you should instead navigate using [`goto`](#modules-$app-navigation).
```
